lib/bloc/backup/backup_bloc.dart

```diff
@@ -491,15 +539,48 @@ class BackupBloc with AsyncActionsHandler {
 
   Future _backupNow(BackupNow action) async {
     try {
-      log.info("backup now requested: $action");
+      log.info("Backup Now requested: $action");
+      final backupProviderName =
+          action.updateBackupSettings.settings.backupProvider.displayName;
+
       await _updateBackupSettings(action.updateBackupSettings);
+      log.info("Does backup service need login $_backupServiceNeedLogin");
       if (_backupServiceNeedLogin) {
+        log.info("Signing out of $backupProviderName");
         await _breezLib.signOut();
+        log.info("Signed out of $backupProviderName");
       }
+      log.info("Signing into $backupProviderName");
       await _breezLib.signIn(_backupServiceNeedLogin, action.recoverEnabled);
+      log.info("Signed into $backupProviderName");
       await _saveAppData();
       await _breezLib.requestBackup();
       action.resolve(true);
+    } on PlatformException catch (e) {
```

Comment:
Can we not handle this only on sign in and throw an error which invokes the "EnableBackupDialog"?